### PR TITLE
qemu: update to 7.2.0

### PR DESCRIPTION
**Pull request by vallon**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.3 21G419 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files? *ran arm qemu virtual machine*
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken? *lots of variants, checked default builds and my custom flavor runs my VM*

### CODE REVIEW

**Comment by raimue** [comment]
Thank you for the port update!

However, I have some notes based on the changelog provided by QEMU upstream.

https://wiki.qemu.org/ChangeLog/7.2

The changelog mentions that libslirp is no longer bundled. I think we might need to use that from a port now to get feature parity with the previous version.

Additionally, it is now possible to build combinations of Cocoa, GTK and SDL UI interfaces into the same binary. Therefore the variants should no longer be conflicting.

**Comment by vallon** [comment]
Yes, I just realized networking is not working.  I see there is libslirp in ports.  I'll try to update.

I don't know enough about the rest to know whether/how to test the variants.  Maybe another PR?

File: emulators/qemu/Portfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ PortGroup legacysupport 1.1
 legacysupport.newest_darwin_requires_legacy 16
 
 name                    qemu
-version                 7.1.0
+version                 7.2.0
 revision                1
```
Comment (herbygillot) [inline emulators/qemu/Portfile:12]: Also `revision` resets to zero whenever the `version` gets updated:

```suggestion
revision                0
```

Comment (vallon) [reply emulators/qemu/Portfile]: fixed

**Comment by vallon** [comment]
Adding libslirp dependency restored default networking behavior (fixed -nic user: network backend 'user' is not compiled into this binary)

File: emulators/qemu/Portfile [hunk 2]
```diff
@@ -25,11 +25,12 @@ homepage                https://www.qemu.org
...
 
-checksums               rmd160  01c82bed8b0219935c00981e16b573004b3802d0 \
-                        sha256  a0634e536bded57cf38ec8a751adb124b89c776fe0846f21ab6c6728f1cbbbe6 \
-                        size    121833004
+checksums               rmd160  50661dd45f91d900241e0e6b919140a6b8a2b2cf \
+                        sha256  5b49ce2687744dad494ae90a898c52204a3406e84d072482a1e1be854eeb2157 \
+                        size    122408576
 
 depends_build           port:texinfo \
+                        port:libslirp \
```
Comment (raimue) [inline emulators/qemu/Portfile:33]: This is good, but we should also add an explicit `--enable-slirp` to the `configure.args` to ensure we always link with libslirp and do not rely on the auto-detection of the configure script.

This is small enough that we do not need it as a variant (user networking is so common that I doubt anyone really needs to deselect it for minimal gains).

**Comment by raimue** [comment]
> I don't know enough about the rest to know whether/how to test the variants. Maybe another PR?

Fine with me. Let's focus on getting the new QEMU 7.2.0 version in with this PR. The no longer conflicting variants can still be resolved later. I did not have a chance to play with building a single binary for multiple GUIs.

**Comment by catap** [comment]
FYI: 8.0.0 was released a few weeks ago

**Comment by raimue** [comment]
Thank you for the pull request. However, this has now been superseded by #19283, which also makes the correct changes to handle the switch to libslirp.

**Pull request closed without merging**